ssl/statem/statem_lib.c

```diff
@@ -1004,6 +1005,73 @@ int ssl_choose_server_version(SSL *s, CLIENTHELLO_MSG *hello)
         break;
     }
 
+    suppversions = tls_get_extension_by_type(hello->pre_proc_exts,
+                                             hello->num_extensions,
+                                             TLSEXT_TYPE_supported_versions);
+
+    /*
+     * TODO(TLS1.3): We only look at this if our max protocol version is TLS1.3
+     * or above. Should we allow it for lower versions too?
```

Comment:
I had a discussion with @davidben, about this. I'm now erring towards ignoring this if server version <1.3. There are some weird things about this. What happens if we get a ClientHello.version == TLS1.2, but max version in supported versions list is TLS1.1? What should the ClientHello.version be set to when using this extension and not using TLS1.3, and what do we do if the extension and supported_versions list are conflicting in some way?